app/models/session.rb

```diff
@@ -4,14 +4,55 @@ class Session < ApplicationRecord
   validates :user_agent, length: { in: 0..255, allow_nil: true }
   validates :api_version, inclusion: { in: %w(20200115) }
 
-  has_secure_token :access_token
-  has_secure_token :refresh_token
+  before_create do
+    set_expiration_dates
+  end
 
-  before_create :extend_expiration_dates
+  SESSION_TOKEN_VERSION = 1
 
   ACCESS_TOKEN_AGE = Rails.application.config.x.session[:access_token_age].seconds
   REFRESH_TOKEN_AGE = Rails.application.config.x.session[:refresh_token_age].seconds
 
+  def self.hash_string(string)
+    Digest::SHA256.hexdigest(string)
+  end
+
+  def create_tokens
+    access_token = SecureRandom.urlsafe_base64
+    refresh_token = SecureRandom.urlsafe_base64
+    self.hashed_access_token = Session.hash_string(access_token)
+    self.hashed_refresh_token = Session.hash_string(refresh_token)
+    save
```

Comment:
I think we should not save here, this method is used on `renew` which is saving also. The same goes for https://github.com/standardnotes/syncing-server/pull/119/files#diff-21301a75c96c49e2bf016f4c63206521R58